polylogue/pipeline.py

```
@@ -128,9 +129,13 @@ def collect_attachments(
                 if needs_download:
                     ok = drive.download_attachment(att_id, local_path)
                     if not ok:
+                        failures.append(att_id)
                         continue
                 drive.touch_mtime(local_path, meta_att.get("modifiedTime"))
             collector.add_local(idx, fname, att_id, local_path)
+    if failures and not dry_run:
+        missing = ", ".join(failures)
+        raise RuntimeError(f"Failed to download attachments: {missing}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)